dev/encoding/vqgan-jax-encoding-yfcc100m-streaming.ipynb

```diff
@@ -0,0 +1,604 @@
+{
```

Comment:
<p>This is a CPU function so we should not try to batch it and instead let <code>datasets</code>  handle it.</p><p> </p><pre class="ql-syntax" spellcheck="false">def prepare_data(example):


&nbsp; &nbsp; key = example['key']
&nbsp; &nbsp; img = example['img']
&nbsp; &nbsp; title = example['title_clean']
&nbsp; &nbsp; description = example['description']


&nbsp; &nbsp; image = get_image(img)
&nbsp; &nbsp; example['image'] = center_crop(image)
&nbsp; &nbsp; example['caption'] = create_caption(title, description)
&nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; return example
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/borisdayma/dalle-mini/pull/84/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
<p>You are right, this is something I wanted to try. This was inherited from a previous test version in which I downloaded URLs using a parallel pool, and that approach benefited from batching. But in this case your suggestion is simpler and better.</p>